File: php/rna-transcription/RnaTranscription.php
<?php

declare(strict_types=1);

function toRna(string $dna): string
{
    // $nucleotides = str_split($dna);
    // $rna = array_map(
    //     fn($item) => strtr($item, ['G'=>'C', 'C'=>'G', 'A'=>'U', 'T'=>'A']),
    //     $nucleotides
    // );
    // return implode($rna);

    return strtr($dna, 'GCAT', 'CGUA');
}

File: go/rna-transcription/rna_transcription.go
package rnatranscription

func ToRNA(dna string) string {
	rna := ""
	for _, nucleotide := range dna {
		rna += dnaNucleotideToRNANucleotide(nucleotide)
	}
	return rna
}

func dnaNucleotideToRNANucleotide(nucleotide rune) string {
	switch nucleotide {
		case 'G':
			return "C"
		case 'C':
			return "G"
		case 'T':
			return "A"
		case 'A':
			return "U"
		default:
			return "?"
	}
}
